### DISCUSSION TITLE
Can i use both Server Side & Client Side translations

### DISCUSSION THREAD

**Discussion by rawand-faraidun** [amannn/next-intl#809] [Q&A]
Sorry, I could not get a better title but I will try to describe it better here.

in my application I have a situation in which I have to use both client translations and server ones too, I have many namespaces like `common`, `home`, and etc..., and I am using them regularly like the documentation. I am so happy that all of my translations won't end up on the client side and I got great SSR and Performance. Still, there is one namespace "statusCode" I have to get into the client side, what I want to know is whether it is possible to have the whole application using the regular way and only provide that namespace messages to the client side for easier use?

maybe by providing that one message into `NextIntlClientProvider` and if I used `const t = useTranslations('statusCode')` it uses the client one so it is available for client components, but any other translations are still done in the server

I hope I could be able to clarify what I need, I can make a provider for that and only add `statusCode` messages and use it there but if it was available from the library itself it would be better.

thanks

**Comment by rawand-faraidun** [comment]
@amannn hey friend, sorry to bother but this question is so important for me

**Reply by amannn** [reply]
Hey, this page should provide answers for this question: https://next-intl-docs.vercel.app/docs/environments/server-client-components

Hope this helps!
